File: ITP1/ITP1_6_C.rb
def generate_official_house
  Array.new(3) do
    ary = []
    10.times { ary << 0 }
    ary
  end
end

def print_output(ridge, delimiter = true)
  ridge.each do |floor|
    puts " " + floor.join(" ")
  end
  puts "#" * 20 if delimiter
end

ridge_1 = generate_official_house
ridge_2 = generate_official_house
ridge_3 = generate_official_house
ridge_4 = generate_official_house

gets.to_i.times do
  b, f, r, v = gets.split(" ").map(&:to_i)

  case b
  when 1
    ridge_1[f-1][r-1] += v
  when 2
    ridge_2[f-1][r-1] += v
  when 3
    ridge_3[f-1][r-1] += v
  when 4
    ridge_4[f-1][r-1] += v
  end
end

print_output(ridge_1)
print_output(ridge_2)
print_output(ridge_3)
print_output(ridge_4, false)

File: ITP1/ITP1_6_C.c
#include<stdio.h>

int main() {
  int ary[4][3][10] = {0};
  int n;
  int i, j, k;
  int a, b, c, d;

  scanf("%d", &n);
  for (i = 0; i < n; i++) {
    scanf("%d %d %d %d", &a, &b, &c, &d);
    ary[a-1][b-1][c-1] += d;
  }

  for (i = 0; i < 4; i++) {
    for (j = 0; j < 3; j++) {
      for (k = 0; k < 10; k++) {
        printf(" %d", ary[i][j][k]);
      }
      printf("\n");
    }
    if ( i <= 2 ) printf("####################\n");
  }

  return 0;
}
